### ISSUE TITLE
Fix memory leak issue when Interprocess fails to listen to messages

### ISSUE THREAD

**Issue by leMicin** [Sidekick-Poe/Sidekick#253]
We should probably limit the reconnection attempts inside the while(true)

Logs from Foldaren

![image](https://github.com/user-attachments/assets/a5563ed4-acae-49cc-9157-d90894963dfc)

**Comment by Wyrade** [comment]
Not sure if it's the same, but opening a second instance of Sidekick Beta causes memory leak in the second instance too (~30MB more gets allocated every sec or less).

I don't think the app should even allow a second instance to open (should either close itself, close the other instance(s), or at least have a setting for it in the settings).

**Comment by leMicin** [comment]
We do have a mechanism to stop multiple instances from running. Can you provide a screenshot of that happening so I can understand better?

**Comment by Wyrade** [comment]
I got an update just now, and currently i don't see a memory leak like before (shooting up to infinity), but it still can start multiple instances.
I just run the sitekick beta shortcut the .application installer put in my start menu, a "starting application" window pops up for a couple of seconds, then i get the window with the "initializing" progress bar. And the same happens if I run the shortcut again, opening another instance, resulting in another icon on the right side of the taskbar and another line in the task manager.
Is that enough or would a (non-english) screenshot actually help?

**Comment by leMicin** [comment]
Good point, I don't know how ClickOnce does it really. The auto-updater is a black box to me.

**Comment by leMicin** [comment]
We've implemented a slower loop for the interprocess message handler, but we need to find a permanent fix

**Comment by leMicin** [comment]
With a new launcher (Velopack) and few mentions of interprocess stuff since, I will close this. Let's create a new issue if this comes back.
